community/modules/scripts/ramble-execute/main.tf

```diff
@@ -15,34 +15,69 @@
  */
 
 locals {
-  commands_content = var.commands == null ? "" : indent(4, yamlencode(var.commands))
+  # This label allows for billing report tracking based on module.
+  labels = merge(var.labels, { ghpc_module = "ramble-execute", ghpc_role = "scripts" })
+}
+
+locals {
+  commands_content = var.commands == null ? "echo 'no ramble commands provided'" : indent(4, yamlencode(var.commands))
 
   execute_contents = templatefile(
     "${path.module}/templates/ramble_execute.yml.tpl",
     {
-      pre_script = ". ${var.spack_path}/share/spack/setup-env.sh && . ${var.ramble_path}/share/ramble/setup-env.sh"
+      pre_script = ". /etc/profile.d/spack.sh && . /etc/profile.d/ramble.sh"
       log_file   = var.log_file
       commands   = local.commands_content
     }
   )
 
-  previous_ramble_runner_content = var.ramble_runner == null ? "" : var.ramble_runner["content"]
-
-  runner_content = <<-EOT
-    ${local.previous_ramble_runner_content}
-    ${local.execute_contents}
-  EOT
+  data_runners = [for data_file in var.data_files : merge(data_file, { type = "data" })]
 
   execute_md5 = substr(md5(local.execute_contents), 0, 4)
+  execute_runner = {
+    type        = "ansible-local"
+    content     = local.execute_contents
+    destination = "ramble_execute_${local.execute_md5}.yml"
+  }
+
+  previous_runners = var.ramble_runner != null ? [var.ramble_runner] : []
+  runners          = concat(local.previous_runners, local.data_runners, [local.execute_runner])
 
-  ramble_execute_runner = {
-    "type"        = "ansible-local"
-    "content"     = local.runner_content
-    "destination" = "ramble_execute_${local.execute_md5}.yml"
+  # Destinations should be unique while also being known at time of apply
+  combined_unique_string = join("\n", [for runner in local.runners : runner["destination"]])
+  combined_md5           = substr(md5(local.combined_unique_string), 0, 4)
+  combined_runner = {
+    type        = "shell"
+    content     = module.startup_script.startup_script
+    destination = "combined_install_spack_${local.combined_md5}.sh"
   }
+
+  bucket_md5  = substr(md5("${var.project_id}.${var.deployment_name}"), 0, 4)
+  bucket_name = "ramble-scripts-${local.bucket_md5}-${local.combined_md5}"
+}
+
+resource "google_storage_bucket" "bucket" {
+  count                       = var.gcs_bucket_path != null ? 0 : 1
```

Comment:
I thought this would be problematic when chaining execute modules. When a resource is conditionally created (such as in this case), it can sometimes cause problems for downstream dependent resources because Terraform cannot build out the resource graph without knowing the outcome of applying the upstream conditional. 

I tried to break this by deploying chained ramble-execute modules, but It deployed without problem. I think all of the work we are doing with hashing instead of using a random num is helping. Feel free to close this comment without action. I just wanted to provide some context in case you see an error along these lines.